drivers/hmis/app/graphql/mutations/ac_hmis/void_referral_request.rb

```diff
@@ -0,0 +1,51 @@
+###
+# Copyright 2016 - 2023 Green River Data Analysis, LLC
+#
+# License detail: https://github.com/greenriver/hmis-warehouse/blob/production/LICENSE.md
+###
+
+module Mutations
+  class AcHmis::VoidReferralRequest < CleanBaseMutation
+    description 'Void a referral request'
+
+    argument :referral_request_id, ID, required: true
+
+    field :record, Types::HmisSchema::ReferralRequest, null: true
+
+    def resolve(referral_request_id:)
+      return error_response('connection not configured') unless HmisExternalApis::AcHmis::LinkApi.enabled?
+
+      request = HmisExternalApis::AcHmis::ReferralRequest.active
+        .viewable_by(current_user)
+        .find_by(id: referral_request_id)
+      return error_response('referral request record not found') unless request
+
+      project = Hmis::Hud::Project.viewable_by(current_user).find(request.project_id)
+      return error_response('project record not found') unless project
+
+      allowed = current_user.permissions_for?(project, [:can_manage_incoming_referrals])
+      return error_response('access denied') unless allowed
```

Comment:
So I see you are following the pattern here of returning ValidationErrors for basically anything going wrong. In other mutations, we even return particular error types for each of these (`:not_allowed` for access issues, `:not_found`, etc). I think the idea was: we may want the frontend to be able to have a more structured understanding of the error, in case it wants to render something different in case of an access error, for example. 

I don't think that was the correct approach though, now that we have Sentry set up, because it means that validation errors returned like this are not sent to Sentry. (We could add logic to send them ofc, but that feels silly).

So I know it's not consistent with the rest of the codebase, but I would prefer each of these to `raise` so that we can track them as unexpected exceptions, which they are.

LMK what you think of that!

Reply:
That sounds great! I'd much prefer to raise exceptions; I'll make that adjustment